### DISCUSSION TITLE
OpenFGA adoption in existing Service

### DISCUSSION THREAD

**Discussion by shisingh-apptio** [openfga/community#184] [Q&A]
Hello Everyone, How are you?

I need your quick suggestion on how to onboard OpenFGA to work with existing API.
I am very beginner with Go(Have good experience with java) and having little trouble understanding it.
If someone can point me any sample github repo where you integrated it to run with any existing service, that would be of great help.

I am using echo to manage server and need to expose APIs which internally will talk to OpenFGA.
Can I make it work without running OpenFGA server as a go-routine? Please help me with the best way forward.

Thanks,
Shivesh

**Comment by aaguiarz** [comment]
Hi Shivesh,

- This example is a very simple one @jon-whit built for an Auth0 FGA presentation, the code would be the same for OpenFGA https://github.com/jon-whit/auth0fga-demo/blob/main/service/service.go
- [This project uses](https://github.com/interline-io/)  OpenFGA but is larger, the direct calls to FGA are here https://github.com/interline-io/transitland-server/blob/main/auth/fga/fga.go

You'll need to run the server as a separate process and call it using [OpenFGA's Go SDK](https://github.com/openfga/go-sdk).

Hope that helps.

**Reply by shisingh-apptio** [reply]
Hi Aaguiarz,thanks for your reply. 
I was mostly looking for the answer how to use OpenFGA as a part of existing server. Its easy to understand if we run it as different service, we can access it using client-sdk or through api but I am having trouble in understanding how to use OpenFGA in existing service itself.
